backend/src/Duxcore.ts

```diff
@@ -0,0 +1,41 @@
+import { config as dtCfg } from "dotenv";
+import { SessionServer } from "./classes/SessionServer";
+import keypair from "keypair";
+import { WebSocketServer } from "./classes/WebSocketServer";
+
+export default class Duxcore {
+  public tmpKeypair = keypair();
+  public sessions: SessionServer = new SessionServer(this);
+  public socketServer: WebSocketServer = new WebSocketServer(this);
+
+  private wsPort = process.env.WS_PORT;
+  private sessionServerAddress = process.env.SESSION_REDIS_ADDRESS;
+
+  constructor() {
+    dtCfg();
+  }
+
+  randStr(length: number): string {
+    let result = "";
+    const chars =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    const charsLength = chars.length;
+
+    for (let i = 0; i < length; i++) {
+      result += chars.charAt(Math.floor(Math.random() * charsLength));
+    }
+
+    return result;
+  }
+
+  start(): Promise<Duxcore> {
+    return new Promise((resolve, reject) => {
+      (async () => {
+        // Not sure if the if is properly binded, but promise executors cannot be async - antipattern
```

Comment:
Pending revision

Reply:
wtf, promise executors absoloutly can be async...

Reply:
The sentence was `not sure if _this_ properly binded, but promise executors cannot be async - antipattern`